lib/src/mock.dart

```diff
@@ -725,15 +726,61 @@ Null _registerMatcher(Matcher matcher, bool capture, {String named}) {
   return null;
 }
 
+/// Information about a stub call verification.
+///
+/// This class is most useful to users in two ways:
+///
+/// * verifying call count, via [called],
+/// * collecting captured arguments, via [captured].
 class VerificationResult {
+  /// List of all arguments captured in real calls.
+  ///
+  /// This list will include any captured default arguments and has no
+  /// structure differentiating the arguments of one call from another. Given
+  /// the following class:
+  ///
+  /// ```dart
+  /// class C {
+  ///   String methodWithPositionalArgs(int x, [int y]) => '';
+  ///   String methodWithTwoNamedArgs(int x, {int y, int z}) => '';
+  /// }
+  /// ```
+  ///
+  /// the following stub calls will result in the following captured arguments:
+  ///
+  /// ```dart
+  /// mock.methodWithPositionalArgs(1);
+  /// mock.methodWithPositionalArgs(2, 3);
+  /// var captured = verify(
+  ///     mock.methodWithPositionalArgs(captureAny, captureAny)).captured;
+  /// print(captured); // Prints "[1, null, 2, 3]"
+  ///
+  /// mock.methodWithTwoNamedArgs(1, y: 42, z: 43);
+  /// mock.methodWithTwoNamedArgs(1, y: 44, z: 45);
+  /// var captured = verify(
+  ///     mock.methodWithTwoNamedArgs(any,
+  ///         y: captureAnyNamed('y'), z: captureAnyNamed('z'))).captured;
+  /// print(captured); // Prints "[42, 43, 44, 45]"
+  /// ```
+  ///
+  /// Named arguments are listed in the order they are captured in, not the
+  /// order in which they were passed.
+  // TODO(srawlins): make this final, in a major release, as a breaking change.
   List<dynamic> captured = [];
+
+  /// The number of calls matched in this verification.
   int callCount;
 
   // Whether the test API mismatch has been checked.
   bool _testApiMismatchHasBeenChecked = false;
 
-  VerificationResult(this.callCount) {
-    captured = List<dynamic>.from(_capturedArgs, growable: false);
+  @Deprecated(
+      'User-constructed VerificationResult is deprecated; this constructor may '
+      'be deleted as early as Mockito 5.0.0')
+  VerificationResult(int callCount) : this._(callCount);
+
+  VerificationResult._(this.callCount) {
+    captured = [..._capturedArgs];
```

Comment:
[nit] Consider assigning in the initializer list (which you'd need to do if it was final)